src/pybind/mgr/ansible/module.py

```diff
@@ -179,18 +186,77 @@ def is_effective(self):
         In the case of Ansible, this will be True if the playbooks has been
         executed succesfully.
 
-        :return Boolean: if the playbook has been executed succesfully
+        :return Boolean: if the playbook/ARS operation has been executed
+                         succesfully
         """
 
-        return self.status == ExecutionStatusCode.SUCCESS
+        return self._status == ExecutionStatusCode.SUCCESS
 
     @property
     def is_errored(self):
-        return self.error
+        return self._status == ExecutionStatusCode.ERROR
 
     @property
-    def is_complete(self):
-        return self.is_errored or (self.is_persistent and self.is_effective)
+    def result(self):
+        return self._result
+
+
+class ARSChangeOperation(AnsibleChangeOperation):
```

Comment:
Is there a reason we need both: `ARSChangeOperation` and `AnsibleChangeOperation`?

Reply:
In the "read" side i have "playbookOperation" ( to deal with playbook executions) and "ARSOperation" ( to deal with read operations over the Ansible runner service). ( this is in the PR for host ls)

I do not know if i would need a call to deal with "write" operations using playbooks,,,, but doing things in this way... the code is prepared to do that. And i keep the same logical classes map.